lnst/Recipes/ENRT/ConfigMixins/LongLivedConnectionsMixin.py

```diff
@@ -0,0 +1,230 @@
+        return ceil(self.params.long_lived_conns / self.params.long_lived_conns_per_ip)
+
+    def calculate_client_connections(self, client_id):
+        client_id += 1  # 0-based index
+        if client_id < self.servers_count:
+            return self.params.long_lived_conns_per_ip
+
+        # remaining connections are handled by last client
+        return (
+            self.params.long_lived_conns
+            - (self.servers_count - 1) * self.params.long_lived_conns_per_ip
+        )
+
+    def _prepare_server(self, receiver, conns_count):
+        server = LongLivedServer(
+            server_ip=receiver.address,
+            server_port=self.params.long_lived_conns_port,
+            connections_count=conns_count,
+        )
+
+        job = receiver.device.netns.prepare_job(server)
+
+        return job
+
+    def _prepare_client(self, generator, receiver, conns_count):
+        client = LongLivedClient(
+            server_ip=receiver.address,
+            server_port=self.params.long_lived_conns_port,
+            client_ip=generator.address,
+            connections_count=conns_count,
+        )
+
+        job = generator.device.netns.prepare_job(client)
+
+        return job
+
+    def wait_agent_for_establish_conns(self, agent, timeout=60):
+        # Yeah, a bit hacky but simplest way of waiting.
+        # The problem is, that LNST doesn't support waiting
+        # for a specific condition at a REMOTE agent.
+
+        # TODO: refactor to use condition wait, when implemented
+        while timeout:
+            job = agent.run("ss | grep ESTAB | wc -l")
+            conns = int(job.stdout)
+            logging.debug(
+                f"Connections established: {conns}/{self.params.long_lived_conns}"
+            )
+            if conns > self.params.long_lived_conns:
+                logging.info("All long-lived connections established")
+                break
+            time.sleep(1)
+            timeout -= 1
+
+    def test_wide_configuration(self):
+        config = super().test_wide_configuration()
+        # L4 can address up to 65535 ports (size of addresable space
+        # defined in long_lived_conns_per_ip) Therefore opening connections
+        # may require multiple IPs.
+        host1, host2 = self.matched.host1, self.matched.host2
+
+        ipv4_addr = interface_addresses(self.params.net_ipv4)
+        ipv6_addr = interface_addresses(self.params.net_ipv6)
+
+        for _ in range(2):
+            # 2 addresses are already assigned by super().test_wide_configuration()
+            # those are used for perf tests. Generator needs to be moved to 3rd address
+            next(ipv4_addr)
+            next(ipv6_addr)
+
+        host1.eth0.down()
+        host2.eth0.down()
+        for host in [host1, host2]:
+            for _ in range(self.servers_count):
+                config.configure_and_track_ip(host.eth0, next(ipv4_addr))
+                config.configure_and_track_ip(host.eth0, next(ipv6_addr))
```

Comment:
since we're splitting the ip addresses into two categories:
1. perf addresses implemented by the base recipe
2. addresses for the long lived connections

we should IMO split them in how we track them in the `config` object as well - this way they mix for how the `generate_perf_endpoints` uses them and so you have to do that weird slicing override, if these ip addresses were split you wouldn't have to do that.

Reply:
Makes sense, thx.